src/coordination/raft_state.cpp

```diff
@@ -145,9 +165,12 @@ auto RaftState::InitRaftServer() -> void {
 
 RaftState::~RaftState() {
   spdlog::trace("Shutting down RaftState for coordinator_{}", coordinator_id_);
-  state_machine_.reset();
-  state_manager_.reset();
-  logger_.reset();
+
+  utils::OnScopeExit const reset_shared_ptrs{[this]() {
```

Comment:
Why? We usually use OnScopeExit only if those things are put at the end, if at the beginning no problem

Reply:
Used OnScopeExit because in case there is no raft_server we want to do reset of those objects, and in case there is raft server we also do want to reset those objects. So I grouped them in one place